tests/integration_tests/tests/agentless_tests/test_snapshot.py

```diff
@@ -253,6 +253,28 @@ def test_v_4_2_restore_snapshot_with_resource_availability(self):
         assert blueprints[2]['id'] == 'blueprint_3' and \
             blueprints[2]['visibility'] == 'global'
 
+    def test_v_4_3_encrypt_secrets_in_restore(self):
+        """
+        Validate the encryption of the secrets values for versions before 4.4.0
+        """
+        snapshot_name = 'snap_4.3.0_with_secrets.zip'
+        snapshot_path = self._get_snapshot(snapshot_name)
+        self._upload_and_restore_snapshot(snapshot_path)
+        secrets = self.client.secrets.list(_include=['key'])
+        assert len(secrets) == 3
+
+        # The secrets values as in the snapshot
+        secret_string = self.client.secrets.get('sec1')
+        secret_file = self.client.secrets.get('sec3')
+        assert secret_string.value == 'top_secret'
+        assert 'test_mail' in secret_file.value
+
+        # Validate update secret, the return value is encrypted
+        updated_secret = self.client.secrets.update('sec2', 'top_top_secret')
+        assert updated_secret.value != 'top_top_secret'
+        updated_secret = self.client.secrets.get('sec2')
+        assert updated_secret.value == 'top_top_secret'
```

Comment:
This doesn't seem to actually check that the data is encrypted anywhere- e.g. by checking that the particular database field is not set to the secret_string.value?

Reply:
Is there a better way to validate the value is encrypted than to check that it's value is different?
I can add this 
``` secret_string = get_storage_manager().get(models.Secret, 'sec1')``` 
``` assert secret_string.value != 'top_secret'```
But still it's by checking that the particular database field is not set to the secret_string.value as you said.

Reply:
As you're retrieving the string again afterwards, you're confirming that it's encrypted and decrypted successfully, which is reasonable.

You could also generate the encrypted string in the test and check that's in the DB, but I think that's going to add more complexity and fragility to the test than it is worth.

However, checking that what's stored in the database is not exactly the secret that was generated does feel necessary- otherwise, the old behaviour (just storing it plaintext) would also pass this test.